types/react-relay/legacy.d.ts

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react';
+
+import {
+    IEnvironment,
+    Variables,
+    CacheConfig,
+    GraphQLTaggedNode,
+    RelayContext,
+    PageInfo,
+    OperationType,
+    _FragmentRefs,
+    _RefType,
+    FetchPolicy,
+} from 'relay-runtime';
+
+export {
+    FragmentRef,
+    RelayPaginationProp,
+    RelayProp,
+    RelayRefetchProp,
+} from './ReactRelayTypes';
+
+import { RelayProp, MappedFragmentProps, RelayRefetchProp, RelayPaginationProp } from './ReactRelayTypes';
+
+export {
+    DataID,
+    DeclarativeMutationConfig,
+    Disposable,
+    // RelayRuntime has two environment exports: one interface, one concrete.
+    IEnvironment as Environment,
```

Comment:
@rbalicki2 it looks like in previous versions of `@types/react-relay`, index.d.ts re-exported the concrete `Environment` from relay-runtime. See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4c251988d62d2f0d7bdbb5b2fcf2d7570f2d1da7/types/react-relay/index.d.ts#L80

Was it intended for this to change in `react-relay/legacy`? Is it expected to be a type error when a function's argument is typed as `Environment` imported from `relay-runtime` and is passed a relay environment typed as  `Environment` imported from `react-relay/legacy` (or `react-relay`)?

```
Argument of type 'Environment' is not assignable to parameter of type 'RelayModernEnvironment'.
  Type 'Environment' is missing the following properties from type 'RelayModernEnvironment': configName, revertUpdate, replaceUpdate
```